### ISSUE TITLE
⌨ Employee dashboard: Dialog pop up collapsed screenreader feedback

### ISSUE THREAD

**Issue by MaddyDaigle** [GCTC-NTGC/gc-digital-talent#12890]
## ⌨ Accessibility Issue

When navigating through job application options on the employee dashboard using a screen reader (VoiceOver), the screen reader announces "dialog pop up collapsed" along with the title of the link. This information is not relevant or helpful to the user. It can be confusing, as it suggests that something is collapsed even though no dialog is currently visible.

### WCAG 2.1 Criteria

WCAG 4.1.2 Name, Role, Value – User interface components must provide meaningful names, roles, and values to assistive technologies without unnecessary or confusing information.

Heuristic: Match Between System and the Real World – The system should use clear and understandable language that matches user expectations.

## 🦋 Expected Behaviour

When a user navigates to the job application links, the screen reader should either:

1. Announce only the link title without mentioning "dialog pop up collapsed," and then notify the user that a dialog pop-up is opened when they select the link.
2. Alternatively, the screen reader could provide clear feedback like "open dialog pop up" to better match user expectations.

## 🕵️ Details

- **Operating System**: macOS
- **Browser**: Chrome
- **Assistive Technology**: Voiceover

**Comment by esizer** [comment]
This one is confusing because it is not actually a link and is a dialog which is collapsed. That is the default behaviour for dialog triggers so this is expected behaviour.

Personally, I would prefer it were a link that goes to a page but we have decided to use dialogs instead so there is not much we can do here 😢

**Comment by esizer** [comment]
> [@esizer](https://github.com/esizer) isn't Maddy saying that our "default behaviour for dialog trigger" is bad? She would prefer it to only read the hyperlink text without describing the state (collapsed) that the dialogue is in. Actually, it should be impossible to reach this text if the dialogue is in any other state, right?

Possibly, we could look into this but the problem is, not all dialogs are modal (our mobile menu is a good example) where the dialog can be open but your focus can leave the dialog so you can encounter the button when the menu is expanded.

Since most of our dialogs are modal, we could reverse this functionality if it is confusing users. 

However, most dialogs just include this even though it is slightly more noise and almost always will be collapsed when it comes to it, it is pretty clear that the button expands something other than just being announced as a button and does not require us to add additional text to signify that.

**Comment by petertgiles** [comment]
Reviewed in refinement:
- this behaviour is subjective but we believe it is helpful to the user
- closing for now but we'll reopen if Fable reports it as well
